client/boot/index.js

```diff
@@ -361,6 +362,20 @@ function reduxStoreReady( reduxStore ) {
 		require( 'lib/rubberband-scroll-disable' )( document.body );
 	}
 
+	/*
+	 * Layouts with differing React mount-points will not reconcile correctly,
+	 * so remove an existing single-tree layout by re-rendering if necessary.
+	 *
+	 * TODO (@seear): React 15's new reconciliation algo may make this unnecessary
+	 */
+	page( '*', function( context, next ) {
+		if ( ! isIsomorphicRoute( context.path ) && previousLayoutIsSingleTree() ) {
```

Comment:
> hope React 15 solves it?

And what I mean by this is that we can hopefully remove this `page()` call altogether when we get React 15.